### PR TITLE
feat: add Connectors to license reference

### PR DESCRIPTION
**Pull request by tmetzke**
## What is the purpose of the change

Add Connectors to the license reference.

## Are there related marketing activities

n/a

## When should this change go live?

No specific date, can go live as soon as possible.

## PR Checklist

- [x] My changes apply to an already released version, and I have added them to the relevant `/versioned_docs` directory, or they are not for an already released version.
- [x] My changes apply to future versions, and I have added them to the main `/docs` directory, or they are not for future versions.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

**Comment by tmetzke** [comment]
@UlrikeLiss, maybe you can have a look here as well to make sure we are compliant from a legal perspective. Thank you! 🙂

**Comment by UlrikeLiss** [comment]
Maybe you can write..
"The Connector SDK, REST Connector, Connector Runtime Docker image, and Connectors Bundle Docker image are licensed under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0). All other Connectors provided by Camunda out-of-the-box are licensed to our users and customers under the same proprietary license as the one defined for Operate, Tasklist, Identity, and Optimize above.
Other than that I am fine

File: versioned_docs/version-8.1/reference/licenses.md [hunk 1]
```diff
@@ -16,6 +16,10 @@ Licenses and license information for Zeebe can be found in the Zeebe project [RE
...
 
+### Connectors
+
+The Connector SDK, REST Connector, Connector Runtime Docker image, and Connectors Bundle Docker image are licensed under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0). All other Connectors provided by Camunda out-of-the-box are licensed to our users and customers under a proprietary license. The proprietary license is the same and therefore has the same implications as the one defined for Operate, Tasklist, Identity, and Optimize above.
```
Comment (tmetzke) [inline versioned_docs/version-8.1/reference/licenses.md:21]:
```suggestion
The Connector SDK, REST Connector, Connector Runtime Docker image, and Connectors Bundle Docker image are licensed under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0). All other Connectors provided by Camunda out-of-the-box are licensed to our users and customers under the same proprietary license as the one defined for Operate, Tasklist, Identity, and Optimize above.
```

**Review by christinaausley — APPROVED** [review]
Very minor adjustment in my latest commit. "Out of the box" should be hyphenated if it is an adjective describing the next word (out-of-the-box Connectors, for example) and should not be hyphenated otherwise  👍

**Pull request merged**